pkg/server/service/loadbalancer/mirror/mirror.go

```diff
@@ -65,6 +79,13 @@ func (m *Mirroring) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 				handler.count++
 				handler.lock.Unlock()
 
+				// prepare request, update body from buffer
+				req, err := preserve.prepareFromBuffer()
+				if err != nil {
+					logger.Error("could not prepare request: %v", err)
+					return
+				}
```

Comment:
instead of repeating that operation for each mirror, couldn't we instead do it before L76, so that all mirrors just rely on one unique copy in memory of the request?

Reply:
We have to prepare the body for each mirror call
```
func TestRequest(t * testing.T)  {
	var (
		req *http.Request
		err error
		rec *httptest.ResponseRecorder
		h http.HandlerFunc
		body = "{}"
	)

	h = handler(body, t)

	req, err = http.NewRequest("POST", "/", strings.NewReader(body))
	assert.NoError(t, err)

	// first call
	rec = httptest.NewRecorder()
	h(rec, req)

	// second call (fail)
	rec = httptest.NewRecorder()
	{ // uncomment this block to pass the test
		//req, err = http.NewRequest("POST", "/", strings.NewReader(body))
		//assert.NoError(t, err)
	}
	h(rec, req)
}

func handler(expected string, t * testing.T) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request)  {
		body, err := ioutil.ReadAll(r.Body)
		assert.NoError(t, err)
		assert.Equal(t, expected, string(body))
	}
}
```

Reply:
I didn't see that you were already doing what I was asking, in the call L58, because prepareFromBuffer does two different things, which is not great imho. We should break things up for clarity. I'll send suggestions soon.